keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlResourceRepository.kt

```diff
@@ -357,6 +359,27 @@ open class SqlResourceRepository(
     }
   }
 
+  override fun triggerResourceRecheck(environmentName: String, application: String) {
+    val resourceUids = sqlRetry.withRetry(READ) {
```

Comment:
I think ideally you'd run the select and update in a transaction so that you don't run the (small) risk of the resources changing under you in between reading and updating.

Reply:
ugh you're right, good point. I'll update this